ipam/ippool_manager.go

```diff
@@ -170,6 +206,21 @@ func (m *IPPoolManager) updateStatusTimestamp() {
 // UpdateAddresses manages the claims and creates or deletes IPAddress accordingly.
 // It returns the number of current allocations.
 func (m *IPPoolManager) UpdateAddresses(ctx context.Context) (int, error) {
+	_, err := m.M3UpdateAddresses(ctx)
+	if err != nil {
+		return 0, err
+	}
+	count, err := m.K8sUpdateAddresses(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return count, nil
+}
+
+// UpdateM3Addresses manages the ipclaims.ipam.metal3.io and creates or deletes IPAddress.ipam.metal3.io accordingly.
+// It returns the number of current allocations.
+func (m *IPPoolManager) M3UpdateAddresses(ctx context.Context) (int, error) {
```

Comment:
Let's keep it private and only export `UpdateAddresses`.
```suggestion
func (m *IPPoolManager) m3UpdateAddresses(ctx context.Context) (int, error) {
```